src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -144,19 +144,29 @@ public User getUser(String email) {
   public int getTotalMessageCount(){
     Query query = new Query("Message");
     PreparedQuery results = datastore.prepare(query);
-    return results.countEntities(FetchOptions.Builder.withLimit(1000));
+    return results.countEntities(FetchOptions.Builder.withDefaults());
```

Comment:
hey Caleb, could you explain the difference between the two and how it changes? I just did a cursory search on `withDefaults()` which creates instance with null values but didn't really get it

Reply:
I think you're right, `withDefaults()` creates a `FetchOptions` object with `null` for all values, including the `limit`. From the documentation, "`limit` is the maximum number of results the query will return", meaning that in the unlikely case that there are more than 1000 messages in our system, only the first 1000 would be returned making this statistic incorrect. It is not explicitly stated, but I'm assuming having `null` as the `limit` would return all the messages with no limit, which is what we want in this case.

Reply:
ah I see, because I had thought null would be like zero or empty so was confused over there, thanks for clarifying!